gpu_direct_rdma_access.c

```diff
@@ -838,6 +855,26 @@ struct rdma_device *rdma_open_device_server(struct sockaddr *addr)
     return NULL;
 }
 
+int rdma_reset_device(struct rdma_device *device)
+{
+    struct ibv_qp_attr      qp_attr;
+    enum ibv_qp_attr_mask   attr_mask;
+    memset(&qp_attr, 0, sizeof qp_attr);
+    /* - - - - - - - Modify QP to RESET - - - - - - - */
+    qp_attr.qp_state = IBV_QPS_RESET;
```

Comment:
probably better to start with modify to ERROR first, make sure all QP WQE's are flushed and then continue to RESET
going to ERROR first will also handle cases that user called this for normal working QP (not already in ERROR)

Reply:
I found interesting comment about this from Dotan Barak:
"Moving a QP to the Reset state will flush all the outstanding WRs and remove them from both Work Queues and Completion Queues.
If you don't need those flushed Work Requests (since they don't contain any critical information for you), you can just move the QP to the Reset state and then to INIT, RTR, RTS state and this will be o.k."
[link](https://www.rdmamojo.com/2013/07/13/how-to-flush-a-work-queue/)

Reply:
from PRM:

29.10.102RST_QP – To Reset QP
The 2RST_QP command transitions a QP from any state to reset state. Once the QP is in reset state, DoorBells are not processed and **WQEs are not flushed by hard-
ware**.

We must have the WQE's flushed, in order to create completions for application. 
This means we need to move to ERROR, wait for all WQE's to be flushed to CQ, and only them move to RST.
This will be a must for the next step in the error handling.
But we can wait with this for next set of commits.